public/app/features/dashboard-scene/scene/new-toolbar/RightActions.tsx

```diff
@@ -44,12 +46,21 @@ export const RightActions = ({ dashboard }: { dashboard: DashboardScene }) => {
 
   const showPanelButtons = isEditingPanel && !hasEditView && !isViewingPanel;
   const showPlayButtons = isPlaying && isShowingDashboard && !isEditingDashboard;
-  const showShareButton = hasUid && !isSnapshot && !isPlaying;
+  const showShareButton = hasUid && !isSnapshot && !isPlaying && !isEditingPanel;
 
   return (
     <ToolbarButtonRow alignment="right" className={styles.container}>
       {renderActionElements(
         [
+          {
+            key: 'dashboard-settings',
+            component: KioskToggleButton,
+            group: '',
+            condition: !isEditingDashboard,
```

Comment:
I think this change should be gated behind the `unifiedNavbars` toggle, and put it back in the topnav when toggle is off.

Reply:
@EdPoole given the "Change theme" we have in profile drawer now , how do you feel about a new entry there "Enter kiosk mode" ? just feels so much better to have this action in a menu vs taking up icon button space ALL the time